package.json

```diff
@@ -1,7 +1,17 @@
 {
   "name": "@iarna/toml",
   "version": "3.0.0",
-  "main": "toml.js",
+  "main": "./toml.js",
+  "exports": {
+    ".": [
+      {
+        "require": "./toml.js",
+        "import": "./toml-esm.mjs"
+      },
+      "./toml.js"
+    ],
+    "./esm": "./toml-esm.mjs"
+  },
```

Comment:
```suggestion
  "exports": {
    ".": {
      "node": {
        "import": "./toml-esm.mjs",
        "require": "./toml.js"
      }
    },
    "./esm": {
      "node": "./toml-esm.mjs"
    }
  },
```

It would be good to [nest](https://nodejs.org/api/esm.html#esm_nested_conditions) these [conditional exports](https://nodejs.org/api/esm.html#esm_conditional_exports) under the `node` condition, in consideration of potential future support for other JS environments like `browser` 🤞 and `deno` (https://github.com/iarna/iarna-toml/issues/30).

Note: I removed the [array fallback](https://nodejs.org/api/esm.html#esm_package_exports_fallbacks). I don't think it's necessary, because it's a feature about supporting "future" specifier syntax (which isn't used here), rather than providing a fallback for legacy Node versions—that's what `package.main` is for.

Reply:
I appreciate the suggestion about nesting conditions and will add it, however the fallback is necessary to avoid errors when loading `package.json` in Node 12. The docs may mention future support as a use case, however that doesn't mean other good use cases don't exist, such as in this case.

Reply:
After looking into it a bit more, it appears that conditional exports have landed in Node 12 at some point. Support for older versions of Node 12 is debatably useful, but using the `node` condition would void support of those versions anyway. Although my personal opinion is that support of older versions is bad practice, the library owner [has a different viewpoint](https://github.com/iarna/iarna-toml/issues/32#issuecomment-653829551), so I'll avoid using the `node` condition as well for now.

Reply:
The rule for backwards compat is just that we shouldn't needlessly break existing users of the library. Introducing new, optional, features, like esm support, need not support all the way back.

Reply:
As far as I'm aware, the error in question is a package.json parsing error. That means existing users not using the feature would also have problems. They don't need to be using ESM.